biz.aQute.resolve/src/biz/aQute/resolve/ProjectResolver.java

```diff
@@ -81,10 +82,14 @@ private String toString(ServiceReference< ? > sr) {
 	private ResolveProcess					resolve		= new ResolveProcess();
 	private Collection<ResolutionCallback>	cbs			= new ArrayList<ResolutionCallback>();
 
-	public ProjectResolver(Project project) {
+	public ProjectResolver(Project project) throws Exception {
 		super(project);
 		getSettings(project);
 		this.project = project;
+
+		Workspace workspace = project.getWorkspace();
+		if (workspace != null)
+			project.addBasicPlugin(new WorkspaceResourcesRepository(workspace));
```

Comment:
Right...

On Aug 31, 2016 12:11 PM, "BJ Hargrave" <notifications@github.com> wrote:

> In biz.aQute.resolve/src/biz/aQute/resolve/ProjectResolver.java
> <https://github.com/bndtools/bnd/pull/1663#discussion_r77017403>:
>
> >  		super(project);
> >  		getSettings(project);
> >  		this.project = project;
> > +
> > +		Workspace workspace = project.getWorkspace();
> > +		if (workspace != null)
> > +			project.addBasicPlugin(new WorkspaceResourcesRepository(workspace));
>
> Also, you did not modify the bnd resolve operation to remove its creation
> of a workspace repository since it is now also done by the ProjectResolver
> constructor.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/bndtools/bnd/pull/1663/files/e24ca8edb8a2cc86c87240547729c03f5819df23#r77017403>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAI9TCGVRsQM0FVNtfslpRYu5k6t1MlKks5qlaZ3gaJpZM4JwIDm>
> .
>